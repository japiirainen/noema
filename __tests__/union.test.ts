import { union, string, number } from "../src/index.js"
import { expectParseFailure, expectParseSuccess } from "./helpers.js"

describe("union", () => {
  test("an empty union should not accept any values", () => {
    expectParseFailure(union([]), "")
  })

  test("should work with a single member", () => {
    expectParseSuccess(union([string]), "")
    expectParseFailure(union([]), 0)
  })

  test("should work with multiple members", () => {
    expectParseSuccess(union([string, number]), "")
    expectParseSuccess(union([string, number]), 0)
    expectParseFailure(union([string, number]), {})
  })
})
