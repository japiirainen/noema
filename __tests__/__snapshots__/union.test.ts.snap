// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`union should work with multiple members 1`] = `
[
  {
    "code": "invalid_union",
    "issues": [
      {
        "code": "invalid_type",
        "message": "Expected value to be a string",
        "path": "",
        "value": {},
      },
      {
        "code": "invalid_type",
        "message": "Expected value to be a number",
        "path": "",
        "value": {},
      },
    ],
    "message": "Invalid union",
    "path": "",
    "value": {},
  },
]
`;
